pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -163,21 +163,27 @@ def _process_stream(msg):
         for key, option in self.options.items():
             try:
                 param = key.split('.')
-                if param[0] == 'mosek':
-                    param.pop(0)
-                param = getattr(mosek, param[0])(param[1])
-                if 'sparam' in key.split('.'):
-                    self._solver_model.putstrparam(param, option)
-                elif 'dparam' in key.split('.'):
-                    self._solver_model.putdouparam(param, option)
-                elif 'iparam' in key.split('.'):
-                    if isinstance(option, str):
-                        option = option.split('.')
-                        if option[0] == 'mosek':
-                            option.pop('mosek')
-                        option = getattr(mosek, option[0])(option[1])
-                    else:
-                        self._solver_model.putintparam(param, option)
+                if key == param[0]:
+                    self._solver_model.putparam(key, option)
+                else:
+                    if param[0] == 'mosek':
+                        param.pop(0)
+                    param = getattr(mosek, param[0])(param[1])
+                    if 'sparam' in key.split('.'):
```

Comment:
I believe the following is both simpler and faster: 
```suggestion
                    if key.endswith('.sparam':
```
(and the same changes in the next 2 tests)

Reply:
Ah, fair point. I have replaced it with `if 'sparam.' in key:` and removed the call to split (the parameter names are things like `mosek.iparam.log`, so can't use endswith)